Use C10_UNUSED instead of (void)X

r-barnes opened the pull request:
Summary:
Auto-generated with
```
buck run //scripts/rbarnes/regex_multiline_replacer:regex_multiline_replacer -- --find '^(\s*for\s*\()(const.*\n)\s*\(void\)[A-Za-z]+;\s*//\s*Suppress.*\s*\n(.*)'  --replace '\1C10_UNUSED \2\3' `find caffe2/ -regex ".*\.\(cpp\|h\)"`
```

Differential Revision: D33432600

albanD reviewed (approved):
SGTM thanks for the cleanup

r-barnes committed:
Use C10_UNUSED instead of (void)X (#70768)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/70768

Auto-generated with
```
buck run //scripts/rbarnes/regex_multiline_replacer:regex_multiline_replacer -- --find '^(\s*for\s*\()(const.*\n)\s*\(void\)[A-Za-z]+;\s*//\s*Suppress.*\s*\n(.*)'  --replace '\1C10_UNUSED \2\3' `find caffe2/ -regex ".*\.\(cpp\|h\)"`
```

Differential Revision: D33432600

fbshipit-source-id: c82c668e79b8be7ede143487cb9f103f33848bdd
